AST ImplicitCastExpr node has optional `part_of_explicit_cast` token

The `ImplicitCastExpr` node has an optional `part_of_explicit_cast` token in Clang 9.0.

# Example

Contents of `g.c`:
```c
int f(int x) {
	return x;
}

int g(unsigned int x) {
	return f((int)x);
}
```

# Panic

```
$ c2go transpile g.c
panic: could not match regexp with string
^(?P<address>[0-9a-fx]+) <(?P<position>.*)> '(?P<type>.*?)'(:'(?P<type2>.*?)')? <(?P<kind>.*)>[\s]*$
0x5581fc12bb10 <col:16> 'unsigned int' <LValueToRValue> part_of_explicit_cast


goroutine 54 [running]:
github.com/elliotchance/c2go/ast.groupsFromRegex(0xc000282620, 0x64, 0xc0001c482e, 0x4d, 0xc0001d3de0)
	/home/u/goget/src/github.com/elliotchance/c2go/ast/ast.go:290 +0x360
github.com/elliotchance/c2go/ast.parseImplicitCastExpr(0xc0001c482e, 0x4d, 0x6b581f)
```

# AST output

The AST of `g.c` looks as follows:

```
$ c2go ast g.c
TranslationUnitDecl 0x5600a80e99f8 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x5600a80ea290 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x5600a80e9f90 '__int128'
|-TypedefDecl 0x5600a80ea300 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x5600a80e9fb0 'unsigned __int128'
|-TypedefDecl 0x5600a80ea5e8 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x5600a80ea3e0 'struct __NSConstantString_tag'
|   `-Record 0x5600a80ea358 '__NSConstantString_tag'
|-TypedefDecl 0x5600a80ea680 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x5600a80ea640 'char *'
|   `-BuiltinType 0x5600a80e9a90 'char'
|-TypedefDecl 0x5600a80ea958 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x5600a80ea900 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x5600a80ea760 'struct __va_list_tag'
|     `-Record 0x5600a80ea6d8 '__va_list_tag'
|-FunctionDecl 0x5600a81487e0 </home/u/Desktop/ccc/g/g.c:1:1, line:3:1> line:1:5 used f 'int (int)'
| |-ParmVarDecl 0x5600a8148710 <col:7, col:11> col:11 used x 'int'
| `-CompoundStmt 0x5600a8148918 <col:14, line:3:1>
|   `-ReturnStmt 0x5600a8148908 <line:2:2, col:9>
|     `-ImplicitCastExpr 0x5600a81488f0 <col:9> 'int' <LValueToRValue>
|       `-DeclRefExpr 0x5600a81488d0 <col:9> 'int' lvalue ParmVar 0x5600a8148710 'x' 'int'
`-FunctionDecl 0x5600a8148a10 <line:5:1, line:7:1> line:5:5 g 'int (unsigned int)'
  |-ParmVarDecl 0x5600a8148948 <col:7, col:20> col:20 used x 'unsigned int'
  `-CompoundStmt 0x5600a8148bc8 <col:23, line:7:1>
    `-ReturnStmt 0x5600a8148bb8 <line:6:2, col:17>
      `-CallExpr 0x5600a8148b90 <col:9, col:17> 'int'
        |-ImplicitCastExpr 0x5600a8148b78 <col:9> 'int (*)(int)' <FunctionToPointerDecay>
        | `-DeclRefExpr 0x5600a8148ab8 <col:9> 'int (int)' Function 0x5600a81487e0 'f' 'int (int)'
        `-CStyleCastExpr 0x5600a8148b28 <col:11, col:16> 'int' <IntegralCast>
          `-ImplicitCastExpr 0x5600a8148b10 <col:16> 'unsigned int' <LValueToRValue> part_of_explicit_cast
            `-DeclRefExpr 0x5600a8148ad8 <col:16> 'unsigned int' lvalue ParmVar 0x5600a8148948 'x' 'unsigned int'
```

--- a/ast/implicit_cast_expr.go
+++ b/ast/implicit_cast_expr.go
@@ -2,12 +2,13 @@ package ast
 
 // ImplicitCastExpr is expression.
 type ImplicitCastExpr struct {
-	Addr       Address
-	Pos        Position
-	Type       string
-	Type2      string
-	Kind       string
-	ChildNodes []Node
+	Addr               Address
+	Pos                Position
+	Type               string
+	Type2              string
+	Kind               string
+	PartOfExplicitCast bool
+	ChildNodes         []Node
 }
 
 // ImplicitCastExprArrayToPointerDecay - constant
@@ -18,17 +19,19 @@ func parseImplicitCastExpr(line string) *ImplicitCastExpr {
 		`<(?P<position>.*)>
 		 '(?P<type>.*?)'
 		(:'(?P<type2>.*?)')?
-		 <(?P<kind>.*)>`,
+		 <(?P<kind>.*)>
+		(?P<part_of_explicit_cast> part_of_explicit_cast)?`,
 		line,
 	)
 
 	return &ImplicitCastExpr{
-		Addr:       ParseAddress(groups["address"]),
-		Pos:        NewPositionFromString(groups["position"]),
-		Type:       groups["type"],
-		Type2:      groups["type2"],
-		Kind:       groups["kind"],
-		ChildNodes: []Node{},
+		Addr:               ParseAddress(groups["address"]),
+		Pos:                NewPositionFromString(groups["position"]),
+		Type:               groups["type"],
+		Type2:              groups["type2"],
+		Kind:               groups["kind"],
+		PartOfExplicitCast: len(groups["part_of_explicit_cast"]) > 0,
+		ChildNodes:         []Node{},
 	}
 }
 

--- a/ast/implicit_cast_expr_test.go
+++ b/ast/implicit_cast_expr_test.go
@@ -36,6 +36,14 @@ func TestImplicitCastExpr(t *testing.T) {
 			Kind:       "LValueToRValue",
 			ChildNodes: []Node{},
 		},
+		`0x5600a8148b10 <col:16> 'unsigned int' <LValueToRValue> part_of_explicit_cast`: &ImplicitCastExpr{
+			Addr:               0x5600a8148b10,
+			Pos:                NewPositionFromString("col:16"),
+			Type:               "unsigned int",
+			Kind:               "LValueToRValue",
+			PartOfExplicitCast: true,
+			ChildNodes:         []Node{},
+		},
 	}
 
 	runNodeTests(t, nodes)

